sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/StaticCheckpointGenerator.java

```diff
@@ -19,18 +19,18 @@
 
 import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkNotNull;
 
-/** Always returns the same instance of checkpoint. */
 class StaticCheckpointGenerator implements CheckpointGenerator {
+  private static final long serialVersionUID = 5972850685627641931L;
```

Comment:
> My fixes included adding serialVersionUID into two classes, and I'm not 100% sure it's actually allowed, can you confirm? 

Not defining a `serialVersionUID` is a poor practice for `Serializable` classes that require it. Unfortunately that's how things started with this IO.  As we might risk changing the default `serialVersionUID` when modifying classes, we should absolutely set it (here, but also for all other classes of this IO that get serialized).

Please make sure to extract it from the latest release artifact (2.46) using the JDK's [serialver](https://docs.oracle.com/javase/8/docs/technotes/tools/unix/serialver.html) tool and document this in a comment.

Note, the resulting `serialVersionUID` is dependent on the class file the compiler produces and not the JDK used as runtime:

> "[I]t is strongly recommended that all serializable classes explicitly declare serialVersionUID values, since the default serialVersionUID computation is highly sensitive to class details that may vary depending on compiler implementations, and can thus result in unexpected InvalidClassExceptions during deserialization. Therefore, to guarantee a consistent serialVersionUID value across different Java compiler implementations, a serializable class must declare an explicit serialVersionUID value."

Reply:
> here, but also for all other classes of this IO that get serialized

Should I add `serialVersionUID` for `KinesisSource` then?